File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.js
/**
 * @param {string} s
 * @return {string}
 */
var removeOuterParentheses = function(s) {
    
    let stack = [];

    return s.split('').reduce((result, char)=>{
        if(char === '(')
        {   
            if(stack.length > 0)
                result += char;
            
            stack.push(char);
        }
        else 
        {
            stack.pop();

            if(stack.length > 0)
                result += char;
        }

        return result;
    },'');
};

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        
        String result = "";
        int count = 0;
        int length = s.length();

        for(int i = 0 ; i < length ; i++)
        {
            char ch = s.charAt(i);

            if( ch == '(')
                count++;
            else
                count--;
            
            if(count != 0  && (count != 1 || (count == 1 && ch == ')')))
            {
                result = result + ch;
            }
        }

        return result;
    }
}